### PR TITLE
Fix for "changing product handle throws 404"

### LINKED ISSUE #4023: Changing handle and hitting publish throw a 404
## Issue Description
If you change the handle of a previously published product and hit publish you get a 404 error

## Steps to Reproduce
1. Create a product and publish
1. Change the title and handle
1. Publish
1. Observe the 404 error

### Versions
```
Node: 8.9.4
NPM: 5.6.0
Meteor Node: 8.9.4
Meteor NPM: 5.6.0
Reaction CLI: 0.29.0
Reaction: 1.10.0
Reaction branch: release-1.10.0
Docker: 17.12.0-ce
```

### PR DESCRIPTION
**Pull request by prinzdezibel**
Resolves #4023   
Impact: **major**  
Type: **bugfix**

## Issue
If you change the handle of a previously published product and hit publish you get a 404 error

## Solution

Re-apply the changes from [PR 3755](https://github.com/reactioncommerce/reaction/pull/3755/files) for file `imports/plugins/core/revisions/server/methods.js` which somehow got lost.

## Breaking changes
None expected

## Testing
1. As an admin, create a product and publish
2. Change the title and handle
3. Publish
4. The URL in the browser's address bar should change and the product should still be displayed

### CODE REVIEW

**Comment by prinzdezibel** [comment]
@Akarshit Could you have a look into this again? Thank you.

**Comment by prinzdezibel** [comment]
@Akarshit  This should be easy to move to the finish line, as this change was already approved earlier.
 It just did not make its way into master for reasons I don't know.

**Comment by prinzdezibel** [comment]
ready to merge.

File: imports/plugins/core/revisions/server/methods.js [hunk 1]
```diff
@@ -188,6 +188,10 @@ Meteor.methods({
     if (revisions) {
       for (const revision of revisions) {
         if (!revision.documentType || revision.documentType === "product") {
+          const oldDocument = Products.findOne(revision.documentId);
+          if (oldDocument && revision.documentData.handle !== oldDocument.handle) {
+            revision.documentData.changedHandleWas = oldDocument.handle;
```
Comment (spencern) [inline imports/plugins/core/revisions/server/methods.js]: `changedHandleWas` is not a good name

**Comment by prinzdezibel** [comment]
@spencern Renamed to `previousHandle`
> Re: Can we do it without touching database?
That would be possible, but would require a canonical way on how we temporarily store data on the server that is specific to the client connection, e.g. server-side session.

**Comment by aldeed** [comment]
@spencern I'm thinking we should probably skip merging this since it will be fixed by the removal of revisions in 1.13.0 anyway. Unless this is important enough to hotfix onto 1.12.0

**Comment by spencern** [comment]
I'm for that as well @aldeed this issue has been complicated because of the revisions system and will be much simpler to solve in 1.13

**Pull request closed without merging**